exporter/trace/jaeger/jaeger_test.go

```diff
@@ -15,24 +15,132 @@
 package jaeger
 
 import (
+	"context"
 	"encoding/binary"
 	"sort"
 	"testing"
 	"time"
 
 	"go.opentelemetry.io/api/key"
 
-	apitrace "go.opentelemetry.io/api/trace"
-
 	"github.com/google/go-cmp/cmp"
+	"github.com/stretchr/testify/assert"
 	"google.golang.org/grpc/codes"
 
+	apitrace "go.opentelemetry.io/api/trace"
+	sdktrace "go.opentelemetry.io/sdk/trace"
+
 	"go.opentelemetry.io/api/core"
 	gen "go.opentelemetry.io/exporter/trace/jaeger/internal/gen-go/jaeger"
 	"go.opentelemetry.io/sdk/export"
 )
 
-// TODO(rghetia): Test export.
+func TestNewExporter(t *testing.T) {
+	const (
+		collectorEndpoint = "http://localhost"
+		serviceName       = "test-service"
+		tagKey            = "key"
+		tagVal            = "val"
+	)
+	// Create Jaeger Exporter
+	exp, err := NewExporter(
+		WithCollectorEndpoint(collectorEndpoint),
+		WithProcess(Process{
+			ServiceName: serviceName,
+			Tags: []core.KeyValue{
+				key.String(tagKey, tagVal),
+			},
+		}),
+	)
+
+	assert.NoError(t, err)
+	assert.EqualValues(t, serviceName, exp.process.ServiceName)
+	assert.Len(t, exp.process.Tags, 1)
+}
+
+func TestNewExporterShouldFailIfCollectorEndpoingEmpty(t *testing.T) {
+	_, err := NewExporter(
+		WithCollectorEndpoint(""),
+	)
+
+	assert.Error(t, err)
+}
+
+type testCollectorEnpoint struct {
+	spansUploaded []*gen.Span
+}
+
+func (c *testCollectorEnpoint) upload(batch *gen.Batch) error {
+	c.spansUploaded = append(c.spansUploaded, batch.Spans...)
+	return nil
+}
+
+var _ batchUploader = (*testCollectorEnpoint)(nil)
+
+func withTestCollectorEndpoint() func() (batchUploader, error) {
+	return func() (batchUploader, error) {
+		return &testCollectorEnpoint{}, nil
+	}
+}
+
+func TestExporter_ExportSpan(t *testing.T) {
+	const (
+		serviceName = "test-service"
+		tagKey      = "key"
+		tagVal      = "val"
+	)
+	// Create Jaeger Exporter
+	exp, err := NewExporter(
+		withTestCollectorEndpoint(),
+		WithProcess(Process{
+			ServiceName: serviceName,
+			Tags: []core.KeyValue{
+				key.String(tagKey, tagVal),
+			},
+		}),
+	)
+
+	assert.NoError(t, err)
+
+	tp, err := sdktrace.NewProvider(
+		sdktrace.WithConfig(sdktrace.Config{DefaultSampler: sdktrace.AlwaysSample()}),
+		sdktrace.WithSyncer(exp))
+
+	assert.NoError(t, err)
+
+	apitrace.SetGlobalProvider(tp)
+	_, span := apitrace.GlobalProvider().GetTracer("test-tracer").Start(context.Background(), "test-span")
+	span.End()
+
+	assert.True(t, span.SpanContext().IsValid())
+
+	exp.Flush()
+	tc := exp.uploader.(*testCollectorEnpoint)
+	assert.True(t, len(tc.spansUploaded) > 0)
```

Comment:
please check for specific count of spans.

Reply:
done